### PR TITLE
whitelist duckdb

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
whitelist duckdb

### Linked issues
Progresses #1901 

### Functionality
None

### Tests
- [x] manually tested

### CODE REVIEW

File: src/databricks/labs/ucx/source_code/known.json [hunk 1]
```diff
@@ -3141,6 +3141,83 @@
     "dockerpycreds.utils": [],
     "dockerpycreds.version": []
   },
+  "duckdb": {
+    "adbc_driver_duckdb": [],
+    "adbc_driver_duckdb.dbapi": [],
+    "duckdb": [
+      {
+        "code": "table-migrate",
+        "message": "Can't migrate 'conn.table(table_name, **kwargs)' because its table name argument cannot be computed"
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3150]: I expect that this is a false positive. It is probably a duckdb connection, not a Spark connection

File: src/databricks/labs/ucx/source_code/known.json [hunk 2]
```diff
@@ -3141,6 +3141,83 @@
...
+      },
+      {
+        "code": "table-migrate",
+        "message": "Can't migrate table_name argument in 'conn.sql(query, **kwargs)' because its value cannot be computed"
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3154]: Same as above

File: src/databricks/labs/ucx/source_code/known.json [hunk 3]
```diff
@@ -3141,6 +3141,83 @@
...
+    "duckdb.experimental.spark.errors.exceptions": [],
+    "duckdb.experimental.spark.errors.exceptions.base": [],
+    "duckdb.experimental.spark.errors.utils": [],
+    "duckdb.experimental.spark.exception": [],
+    "duckdb.experimental.spark.sql": [],
+    "duckdb.experimental.spark.sql._typing": [],
+    "duckdb.experimental.spark.sql.catalog": [
+      {
+        "code": "table-migrate",
+        "message": "Can't migrate table_name argument in 'self._session.conn.sql(query)' because its value cannot be computed"
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3175]: I expect this also to be a false positive, but should be verified more thoroughly due to the `spark` in the module name. DuckDB mimics the Spark API, still I expect this to be only duckdb

File: src/databricks/labs/ucx/source_code/known.json [hunk 4]
```diff
@@ -3141,6 +3141,83 @@
...
+      }
+    ],
+    "duckdb.experimental.spark.sql.column": [],
+    "duckdb.experimental.spark.sql.conf": [],
+    "duckdb.experimental.spark.sql.dataframe": [],
+    "duckdb.experimental.spark.sql.functions": [],
+    "duckdb.experimental.spark.sql.group": [],
+    "duckdb.experimental.spark.sql.readwriter": [
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3183]: Also false positive

File: src/databricks/labs/ucx/source_code/known.json [hunk 5]
```diff
@@ -3141,6 +3141,83 @@
...
+      {
+        "code": "table-migrate",
+        "message": "Can't migrate table_name argument in 'self.session.conn.sql(f'select * from {path}')' because its value cannot be computed"
+      }
+    ],
+    "duckdb.experimental.spark.sql.session": [
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3189]: This too

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
See comments

File: src/databricks/labs/ucx/source_code/known.json [hunk 6]
```diff
@@ -3141,6 +3141,83 @@
...
+      },
+      {
+        "code": "table-migrate",
+        "message": "Can't migrate table_name argument in 'conn.sql(query, **kwargs)' because its value cannot be computed"
+      }
+    ],
+    "duckdb.bytes_io_wrapper": [],
+    "duckdb.experimental": [],
+    "duckdb.experimental.spark": [],
+    "duckdb.experimental.spark._globals": [],
```
[the remark this answers is not in the document]
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3160]: They were merged so I guess the decision is to keep them ?

File: src/databricks/labs/ucx/source_code/known.json (hunk 3 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3175]: fixed

File: src/databricks/labs/ucx/source_code/known.json (hunk 4 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3183]: fixed

File: src/databricks/labs/ucx/source_code/known.json (hunk 5 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json]: fixed

File: src/databricks/labs/ucx/source_code/known.json (hunk 1 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3150]: fixed

File: src/databricks/labs/ucx/source_code/known.json (hunk 2 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3154]: fixed

File: src/databricks/labs/ucx/source_code/known.json [hunk 7]
```diff
@@ -3141,6 +3141,83 @@
     "dockerpycreds.utils": [],
     "dockerpycreds.version": []
   },
+  "duckdb": {
+    "adbc_driver_duckdb": [],
```
[the remark this answers is not in the document]
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3145]: it's the duckdb driver which complies with the abdc_driver stack, not sure what potential issue you have in mind ?

[the remark this answers is not in the document]
Comment (JCZuurmond) [reply src/databricks/labs/ucx/source_code/known.json:3145]: the pattern is that the key always matches the top level module name, so I don't know if the code is able to handle that it does not match here. Line 3144 `duckdb` does not match the start of the key in line 3145

File: src/databricks/labs/ucx/source_code/known.json (hunk 3 above)
Comment (JCZuurmond) [reply src/databricks/labs/ucx/source_code/known.json:3175]: The `table-migrate` is still there

File: src/databricks/labs/ucx/source_code/known.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3145]: I confirm that the code handles that. The top-level key is only used to build known.json, every key under that is a full-fledged module key.

File: src/databricks/labs/ucx/source_code/known.json [hunk 8]
```diff
@@ -3187,6 +3187,48 @@
...
+    "duckdb.experimental.spark.sql": [],
+    "duckdb.experimental.spark.sql._typing": [],
+    "duckdb.experimental.spark.sql.catalog": [],
+    "duckdb.experimental.spark.sql.column": [],
+    "duckdb.experimental.spark.sql.conf": [],
+    "duckdb.experimental.spark.sql.dataframe": [],
+    "duckdb.experimental.spark.sql.functions": [],
+    "duckdb.experimental.spark.sql.group": [],
+    "duckdb.experimental.spark.sql.readwriter": [],
+    "duckdb.experimental.spark.sql.session": [
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/known.json:3216]: Is a false positive, there is no databricks runtime in duckdb. (https://github.com/duckdb/duckdb/blob/33b44ab04fd25034fd89fb33b52261d286704b01/tools/pythonpkg/duckdb/experimental/spark/sql/session.py)

Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3216]: I don't thins it's a false positive. If DuckDB is running within a DBR, then calling DuckDB on parquet storage managed by DBR will fail.

Comment (JCZuurmond) [reply src/databricks/labs/ucx/source_code/known.json:3216]: Could you link the line that this message is pointing to? It is missing in the `known.json`

Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/known.json:3216]: It's probably this code:
https://github.com/duckdb/duckdb/blob/33b44ab04fd25034fd89fb33b52261d286704b01/tools/pythonpkg/duckdb/experimental/spark/sql/session.py#L217

**Review by nfx — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/source_code/known.json (hunk 8 above)
Comment (JCZuurmond) [reply src/databricks/labs/ucx/source_code/known.json:3216]: That is creating a table using the `DuckDBPyConnection`. Still not expect that to be influence by the Databricks runtime. I expect none of the duckdb modules to result a lint message

**Pull request merged**